fedimint-cli/src/lib.rs

```diff
@@ -359,6 +364,35 @@ enum AdminCmd {
     GuardianConfigBackup,
 }
 
+#[derive(Debug, Clone, Subcommand)]
+enum DecodeType {
+    /// Decode an invite code string into a JSON representation
+    InviteCode { invite_code: InviteCode },
+    /// Decode a string of ecash notes into a JSON representation
+    Notes { notes: OOBNotes },
+}
+
+#[derive(Debug, Clone, Subcommand)]
+enum EncodeType {
+    /// Encode connection info from its constituent parts
+    InviteCode {
+        #[clap(long = "url")]
+        url: SafeUrl,
+        #[clap(long = "federation_id")]
+        federation_id: FederationId,
+        #[clap(long = "peer")]
+        peer: PeerId,
+    },
+
+    /// Encode a JSON string of notes to an ecash string
+    Notes {
+        #[clap(long = "federation_id_prefix")]
+        federation_id_prefix: String,
+        #[clap(long = "notes")]
+        notes: String,
```

Comment:
~Shouldn't this be a `Vec<_>`? Ideally a `Vec<SpendableNote>`, but due to https://github.com/clap-rs/clap/issues/1704 likely rather a `Vec<String>`.~ Maybe supplying the whole vec as JSON is actually more convenient.

Reply:
I originally had it as a vec but it's super tedious to input that way, doing it as Json string you can just wrap the JSON output from the dev decode command with single quotes like `'{"notes": stuff}'`